src/Database/Migrations/2020-12-28-223112_create_auth_tables.php

```diff
@@ -57,6 +57,7 @@ public function up(): void
             'id'         => ['type' => 'int', 'constraint' => 11, 'unsigned' => true, 'auto_increment' => true],
             'ip_address' => ['type' => 'varchar', 'constraint' => 255],
             'user_agent' => ['type' => 'varchar', 'constraint' => 255, 'null' => true],
+            'id_type'    => ['type' => 'varchar', 'constraint' => 255],
```

Comment:
This should be added to the `identifier` key since reverse lookup will always require the ID type now too.

Reply:
When and why do we need reverse lookup?

Table `auth_identities` already has `type`: `email_password`, `magic-link`, `access_token`
Use this as `id_type`?

Reply:
Changed to use `auth_identities` type.

Replaced strings with constants In the source code. https://github.com/codeigniter4/shield/pull/181/commits/2464affbf2466ed50ba24ee895364c4a11ae90c7

Reply:
I'm not sure where the reverse lookup happens but we already have the index for it:
```php
$this->forge->addKey('identifier');
```
Since this is not a unique field we theoretically could have two of the same identifiers for two different ID types, so the type should always be used with the identifier for reverse lookup - which makes the current index insufficient.

Reply:
I got what you mean!

Done.